contracts/Aavegotchi/facets/AavegotchiLendingFacet.sol

```diff
@@ -0,0 +1,277 @@
+        }
+        assembly {
+            mstore(rentals_, listIndex)
+        }
+    }
+
+    ///@notice Query a certain amount of aavegotchi rentals
+    ///@param _length How many rentals to return
+    ///@return rentals_ An array of rental
+    function getAavegotchiRentals(uint256 _length) external view returns (AavegotchiRental[] memory rentals_) {
+        uint256 rentalId = s.aavegotchiRentalHead;
+        rentals_ = new AavegotchiRental[](_length);
+        uint256 listIndex;
+        for (; rentalId != 0 && listIndex < _length; listIndex++) {
+            rentals_[listIndex] = s.aavegotchiRentals[rentalId];
+            rentalId = s.aavegotchiRentalListItem[rentalId].childRentalId;
+        }
+        assembly {
+            mstore(rentals_, listIndex)
+        }
+    }
+
+    function isAavegotchiLent(uint256 _erc721TokenId) external view returns (bool) {
+        return LibAavegotchiLending.isAavegotchiLent(_erc721TokenId);
+    }
+
+    ///@notice Allow an original aavegotchi owner to add request for rental
+    ///@dev If the rental request exist, cancel it and replaces it with the new one
+    ///@dev If the rental is active, unable to cancel
+    ///@param _erc721TokenId The identifier of the NFT to rent
+    ///@param _initialCost The rental fee of the aavegotchi in $GHST
+    ///@param _period The rental period of the aavegotchi, unit: second
+    ///@param _revenueSplit The revenue split of the rental, 3 values, sum of the should be 100
+    ///@param _receiver The 3rd account for receive revenue split, can be address(0)
+    ///@param _whitelistId The identifier of whitelist for agree rental, if 0, allow everyone
+    function addAavegotchiRental(
+        uint256 _erc721TokenId,
+        uint256 _initialCost,
+        uint256 _period,
+        uint256[3] calldata _revenueSplit,
+        address _receiver,
+        uint256 _whitelistId,
+        address[] calldata _includes
+    ) external {
+        IERC721 erc721Token = IERC721(address(this));
+        address sender = LibMeta.msgSender();
+        require(erc721Token.ownerOf(_erc721TokenId) == sender, "AavegotchiLending: Not owner of aavegotchi");
+        require(
+            erc721Token.isApprovedForAll(sender, address(this)) || erc721Token.getApproved(_erc721TokenId) == address(this),
+            "AavegotchiLending: Not approved for transfer"
+        );
+
+        require(_period > 0, "AavegotchiLending: period should be larger than 0");
+        //        require(_revenueSplit.length == 3, "AavegotchiLending: revenues split should consists of 3 values");
+        require(_revenueSplit[0] + _revenueSplit[1] + _revenueSplit[2] == 100, "AavegotchiLending: sum of revenue split should be 100");
+        if (_receiver == address(0)) {
+            require(_revenueSplit[2] == 0, "AavegotchiLending: revenue split for invalid receiver should be zero");
+        }
+        require(s.whitelists.length > _whitelistId, "AavegotchiLending: whitelist not found");
```

Comment:
This could be changed to:
`require(s.whitelists.length > _whitelistId || _whitelistId == 0, "AavegotchiLending: whitelist not found");` in the case that no whitelists exist yet.